core/src/main/scala/com/thenewmotion/mobilityid/checkDigit.scala

```diff
@@ -75,5 +77,44 @@ private[mobilityid] object MatrixUtil {
   val negP2minus15 = Matrix(0, 2, 2, 1) // -p2^(-15)
 }
 
+private[mobilityid] object CheckDigitDin {
+
+  private val lookup = {
+    val left = ('0' to '9') ++ ('A' to 'Z')
+    val right = 0 to 35
+
+    (left zip right).toMap
+  }
+
+  def apply(contractId: String): Char = {
+    def mult(value: Int, coeff: Int) = value * math.pow(2, coeff).toInt
```

Comment:
This algorithm must have been designed with `value * (1 << coeff)` in mind. But yes, CPUs have floating point units nowadays.